pkg/csi/service/vanilla/controller.go

```diff
@@ -514,53 +458,33 @@ func (c *controller) ReloadConfiguration() error {
 		}
 		if newVCConfig != nil {
 			newVCConfig.ReloadVCConfigForNewClient = true
+			if oldvCenter != newVCConfig.Host {
+				log.Infof("Observed new vCenter server: %q in the config secret. "+
+					"Exiting vSphere CSI Controller Container for re-initialization", newVCConfig.Host)
+				unregisterAllvCenterErr := c.managers.VcenterManager.UnregisterAllVirtualCenters(ctx)
+				if unregisterAllvCenterErr != nil {
+					log.Warnf("failed to Unregister all vCenter servers. Error: %v. "+
+						"Proceeding to exit the vSphere CSI Controller Container for re-initialization",
+						unregisterAllvCenterErr)
+				}
+				os.Exit(1)
+			}
+
 			var vcenter *cnsvsphere.VirtualCenter
 			vcenter, err = cnsvsphere.GetVirtualCenterInstance(ctx, &cnsconfig.ConfigurationInfo{Cfg: newCfg}, false)
 			if err != nil {
 				return logger.LogNewErrorf(log, "failed to get VirtualCenter. err=%v", err)
 			}
 			vcenter.Config = newVCConfig
-			var operationStore cnsvolumeoperationrequest.VolumeOperationRequest
-			operationStore, err = cnsvolumeoperationrequest.InitVolumeOperationRequestInterface(ctx,
-				newCfg.Global.CnsVolumeOperationRequestCleanupIntervalInMin,
-				func() bool {
-					return commonco.ContainerOrchestratorUtility.IsFSSEnabled(ctx, common.BlockVolumeSnapshot)
-				})
-			if err != nil {
-				log.Errorf("failed to initialize VolumeOperationRequestInterface with error: %v", err)
-				return err
-			}
-			if oldvCenter != newVCConfig.Host {
-				unregisterErr := c.manager.VcenterManager.UnregisterVirtualCenter(ctx, oldvCenter)
-				if unregisterErr != nil {
-					log.Warnf("failed to Unregister VirtualCenter: %q. Error: %v", c.manager.VcenterConfig.Host, unregisterErr)
-				}
-				authmanager, err := common.GetNewAuthorizationService(ctx, vcenter)
-				if err != nil {
-					log.Errorf("failed to initialize authorization service for vCenter:%q with error: %v",
-						vcenter.Config.Host, err)
-					return err
-				}
-				c.authMgr.Stop()
-				go common.ComputeDatastoreMapForBlockVolumes(authmanager, newCfg.Global.CSIAuthCheckIntervalInMin)
-				c.authMgr = authmanager
-			} else {
-				if c.authMgr != nil {
-					c.authMgr.ResetvCenterInstance(ctx, vcenter)
-					log.Debugf("Updated vCenter in auth manager")
-				}
+			if c.authMgr != nil {
+				c.authMgr.ResetvCenterInstance(ctx, vcenter)
+				log.Info("Updated vCenter in auth manager")
 			}
 			err = c.manager.VolumeManager.ResetManager(ctx, vcenter)
 			if err != nil {
 				return logger.LogNewErrorf(log, "failed to reset volume manager. err=%v", err)
 			}
 			c.manager.VcenterConfig = newVCConfig
-			volumeManager, err := cnsvolume.GetManager(ctx, vcenter, operationStore, true,
-				false, false, tasksListViewEnabled)
-			if err != nil {
-				return logger.LogNewErrorf(log, "failed to create an instance of volume manager. err=%v", err)
-			}
-			c.manager.VolumeManager = volumeManager
```

Comment:
Hey @divyenpatel, trying to understand the changes in #2221 and #2195. Can you please explain why this call to GetManager() was removed?

Reply:
We are restarting CSI controller and Syncer container so we do not need to worry about updating several services for new config changes. This is a much simple approach then updating lot of cached config.